image-prune

```diff
@@ -199,6 +202,30 @@ class LocalImageDirectory(ImageCache):
         return free >= self.space_threshold
 
 
+class S3ImageStore(ImageCache):
+    def __init__(self, url, max_gb=80):
+        self.url = url
+        self.max_bytes = (max_gb * 1000 * 1000 * 1000)
+
+    def list_files(self):
+        result = s3.list_bucket(self.url)
+
+        # make sure this gets done before we start iterating
+        self.sizes = dict(s3.parse_list(result, "Key", "Size"))
+
+        RFC3339ish = '%Y-%m-%dT%H:%M:%S.%f%z'
+        for name, stamp in s3.parse_list(result, "Key", "LastModified"):
+            yield name, datetime.datetime.strptime(stamp, RFC3339ish).timestamp()
+
+    def delete_file(self, filename):
+        with s3.urlopen(self.url._replace(path=filename), method='DELETE') as _:
+            del self.sizes[filename]
```

Comment:
That's certainly a rather curious syntax. I take it you take the `with` just to get an automatic closing? I suppose putting `pass` into the `with` and having the `del` separate is a bit cleaner. But feel free to ignore, that's just bikeshedding.

Reply:
I'm happy to do that.

Also: I learned, meanwhile, that `as _` is optional, and if you exclude it, then the expected closing will happen.